### PR TITLE
publish: initial add

### PR DESCRIPTION
**Pull request by johannesring**

### CODE REVIEW

File: recipes/publish/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{% set version = "1.0.1" %}
+{% set hg_tag = "164:b7f90dcfb235" %}
+
+package:
+  name: publish
+  version: {{ version }}
+
+source:
+  hg_url: https://bitbucket.org/logg/publish
+  hg_tag: {{ hg_tag }}
```
Comment (jakirkham) [inline recipes/publish/meta.yaml]: Please use a tarball.

File: recipes/publish/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+build:
+  number: 2
```
Comment (jakirkham) [inline recipes/publish/meta.yaml]: Build number should start with `0`.

File: recipes/publish/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+about:
+  home: https://bitbucket.org/logg/publish
+  license: GPL-3
+  license_file: COPYING
+  summary: 'publish: a bibliographic reference system'
+
+extra:
+  recipe-maintainers:
+  - johannesring
```
Comment (jakirkham) [inline recipes/publish/meta.yaml]: Please indent.

File: recipes/publish/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  run:
+  - python
+  - lxml
+  - future
+  - python-levenshtein
+
+test:
+  imports:
+  - publish
```
Comment (jakirkham) [inline recipes/publish/meta.yaml]: Please indent.

File: recipes/publish/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  - python
+  - mercurial
+  - future
+  - python-levenshtein
+
+  run:
+  - python
+  - lxml
+  - future
+  - python-levenshtein
```
Comment (jakirkham) [inline recipes/publish/meta.yaml]: Please indent.

File: recipes/publish/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+build:
+  number: 2
+  script: python setup.py install
+
+requirements:
+  build:
+  - python
+  - mercurial
+  - future
+  - python-levenshtein
```
Comment (jakirkham) [inline recipes/publish/meta.yaml]: Please indent. Also, please drop everything except `python`.

File: recipes/publish/meta.yaml (hunk 1 above)
Comment (johannesring) [inline recipes/publish/meta.yaml]: I would if there was a tarball. What should I do when there is no tarball?

Comment (jakirkham) [inline recipes/publish/meta.yaml]: Ask upstream for tags. Use PyPI.

File: recipes/publish/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+build:
+  number: 0
+  script: python setup.py install
+
+requirements:
+  build:
+    - python
+    - mercurial
+    - future
+    - python-levenshtein
```
Comment (jakirkham) [inline recipes/publish/meta.yaml]: We should be able to drop `future` and `python-levenshtein`.

Comment (jakirkham) [inline recipes/publish/meta.yaml]: (from `build`)

File: recipes/publish/meta.yaml (hunk 1 above)
Comment (johannesring) [inline recipes/publish/meta.yaml]: I have asked upstream to make a tag.

Comment (johannesring) [inline recipes/publish/meta.yaml]: Here is the bug report I made about this: https://bitbucket.org/logg/publish/issues/5/please-tag-the-current-version

**Comment by johannesring** [comment]
@jakirkham - upstream has now made a tag and I have updated the recipe. Now the travis tests fail with the following error:

``` console
===== testing package: publish-1.0.1-py34_0 =====
import: 'publish'
Traceback (most recent call last):
  File "/Users/travis/miniconda/conda-bld/test-tmp_dir/run_test.py", line 26, in <module>
    import publish
  File "/Users/travis/miniconda/envs/_test/lib/python3.4/site-packages/publish/__init__.py", line 2, in <module>
    from publish.exporting import export_file
  File "/Users/travis/miniconda/envs/_test/lib/python3.4/site-packages/publish/exporting.py", line 16, in <module>
    from .formats import bibtex, latex, pub, pdf, html, rst, graphml
  File "/Users/travis/miniconda/envs/_test/lib/python3.4/site-packages/publish/formats/graphml.py", line 14, in <module>
    import lxml.etree as xml
ImportError: dlopen(/Users/travis/miniconda/envs/_test/lib/python3.4/site-packages/lxml/etree.so, 2): Library not loaded: @rpath/libxml2.2.dylib
  Referenced from: /Users/travis/miniconda/envs/_test/lib/python3.4/site-packages/lxml/etree.so
  Reason: Incompatible library version: etree.so requires version 12.0.0 or later, but libxml2.2.dylib provides version 10.0.0
TESTS FAILED: publish-1.0.1-py34_0
```
This looks like it might be a bug in the `lxml` package?

**Comment by johannesring** [comment]
@jakirkham, @scopatz - Any idea what is happening here? Looks like it is trying to build with Python 3.6.

**Comment by scopatz** [comment]
Yes, it is. If you don't want to build with python 3.6, you have to skip that case.

**Comment by johannesring** [comment]
Thanks for your reply @scopatz. I'm okay with building with Python 3.6 but it looks like for example one of the requirements `python-levenshtein` has not been built for Python 3.6. I guess I just have to skip the build for Python 3.6 for now.

**Comment by scopatz** [comment]
Or you could open a PR on the levenshtein feedstock that rerenders that for python v3.6.  That is best for everyone.

**Comment by johannesring** [comment]
Yes, good idea. I will do that tomorrow.

**Comment by johannesring** [comment]
@conda-forge-admin - looks like one of the checks is hanging. Can someone restart the build?

**Comment by scopatz** [comment]
It seems like a fair part of the queue is stuck and doesn't give us the ability to cancel

**Comment by johannesring** [comment]
@conda-forge/staged-recipes - the error on Travis does not look like it is related to this PR. Any idea how to fix this?

**Comment by pmlandwehr** [comment]
Seen this issue in some other builds too.

**Comment by johannesring** [comment]
Thanks @pmlandwehr - I see the same issue in at least PR #2486, #2467 and #2475. Unfortunately, no solution in any of them.

**Comment by jakirkham** [comment]
It was an issue in the fast finish script that has been fixed. Restarted Travis CI here.

**Comment by johannesring** [comment]
Thanks @jakirkham - seems like this is finally ready to go in. :-)

**Comment by johannesring** [comment]
@conda-forge/staged-recipes - please merge this when you have time. :-)

**Comment by johannesring** [comment]
Thanks @ocefpaf - feels good to finally get this in after so many months. :-)

**Comment by ocefpaf** [comment]
> Thanks @ocefpaf - feels good to finally get this in after so many months. :-)

We rely on volunteers to review the recipes here and we are just a few for now. Consider applying to be a reviewer to help us out :wink: 

PS: feel free to ping me on gitter to help with reviews and merges here.

**Comment by johannesring** [comment]
Sure @ocefpaf - maybe I will some day. :wink:

**Comment by scopatz** [comment]
Today is the best day :)

**Pull request merged**